Total work of the backend team

FannaIQ opened the pull request:
The algorithm now works for the given input
We still have to use the words provided
we still have to make use if the word is common or not
we still have to implement the python translator for a .txt file
we still have to make use of a lemmatizer

FannaIQ committed:
Total work of the backend team
The algorithm now works for the given input
We still have to use the words provided
we still have to make use if the word is common or not
we still have to implement the python translator for a .txt file
we still have to make use of a lemmatizer

FannaIQ committed:
also needed for the algorithm

AntonioBarral committed:
Context algorithm

Needs to decide how will be the input and output yet.

AntonioBarral committed:
Merge pull request #12 from Nickborchers/backendAntonio

Context algorithm

Raque10bar committed:
Word lists, letter scores and code updated

Angelo404 commented on backend/ContextAlgorithm/ContextAlgorithm.py:
```diff
@@ -0,0 +1,133 @@
+class Sentence:
+
+
+    def __init__(self, sentenceString, words):
```
Having the variable type in the variable name is a good practice in python but Camel Case is not. Separate the different parts for the variable name with underscore

Angelo404 commented on backend/ContextAlgorithm/ContextAlgorithm.py:
```diff
@@ -0,0 +1,133 @@
+        self.difficulty = 0
+
+
+    def addValues(self, values):
```
Again, separate with underscore

Angelo404 commented on backend/ContextAlgorithm/ContextAlgorithm.py:
```diff
@@ -0,0 +1,133 @@
+INPUT_LANG = 'es'
+OUTPUT_LANG = 'en'
+WORD = 'cama'
+WORD_DIFFICULTY = 882
```
Why that specific number?

AntonioBarral replied:
This is because I don't have the input yet. That number is the position of "cama" in the frecuency list. When I'll define the input, WORD, WORD_DIFFICULTY and INPUT_LANG will have the value of the input values

Angelo404 replied:
Ok if that will change in the future its fine, else put a comment explaining that

Angelo404 commented on backend/ContextAlgorithm/ContextAlgorithm.py:
```diff
@@ -0,0 +1,133 @@
+    for x in range(0, len(wordsValues)):
+        error += wordsValues[x] - WORD_DIFFICULTY
+
+    error /= len(wordsValues)
```
You can directly return this line

Angelo404 replied:
If you care about decimal points and you are using python version < python 3 then with this expression you are losing precision

Angelo404 replied:
Depends how much you care

Angelo404 commented on backend/ContextAlgorithm/ContextAlgorithm.py:
```diff
@@ -0,0 +1,133 @@
+def betterSentences(possibleSentences):
+    challengingSentence = ""
+    simplestSentence = ""
+    min = float("-inf")
```
min and max are keywords in python. They should not be used for variable names

Angelo404 commented on backend/ContextAlgorithm/ContextAlgorithm.py:
```diff
@@ -0,0 +1,133 @@
+            max = s.difficulty
+            simplestSentence = s.sentenceString
+
+    return challengingSentence, simplestSentence
```
When you return more than one variable in python return a dictionary. So this line (line 92 currently) should be like this:
{"chall_sntnc": challengingSentence, "smpl_sntnc": simplestSentence}

Angelo404 commented on backend/ContextAlgorithm/ContextAlgorithm.py:
```diff
@@ -0,0 +1,133 @@
+
+
+def main():
+    resp = requests.get('https://linguee-api.herokuapp.com/api?q=' + WORD + '&src=' + INPUT_LANG + '&dst=' + OUTPUT_LANG)
```
This is too long break it into multiple lines

Angelo404 commented on backend/wordAlgorithm/algorithm.py:
```diff
@@ -0,0 +1,57 @@
+    
+    for j in range (1, lenTranslation):
+        for i in range (1, lenOrignalWord):
+            if (originalWord[i-1] == translation[j-1]):
```
Fun fact: you can do this: cost = int(originalWord[i-1] == translation[j-1])

Angelo404 commented on backend/wordAlgorithm/word.py:
```diff
@@ -0,0 +1,94 @@
+words = []
+
+# Word object
+class Word(object):
```
_This is debatable_ Word class i think should contain some functionality. In such classes __str__ function is important to begin with. Also depending on how you use it you might want to add some functions to extend the list:extensions or the int:workDifficulty

Angelo404 commented on backend/wordAlgorithm/word.py:
```diff
@@ -0,0 +1,94 @@
+    wordDifficulty = 0;
+    
+# Word constructor
+def makeWord(language, originalWord, translatedWord, category, extensions, wordDifficulty):
```
Again, dont use Camel Case instead use underscores. and please follow [this commenting convention](https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings).

Angelo404 replied:
IMPORTANT: Just because you can comment everything doesn't mean you should. The very simple functions are not expected to have comments ALTHOUGH i would like to at least contain what type the variables/parameter are (see comment on previous file)

Angelo404 commented on backend/wordAlgorithm/word.py:
```diff
@@ -0,0 +1,94 @@
+
+# Returns the value if
+def checkExistance (value):
+    if not (value is None):
```
Fun fact: this can be done in one line

Angelo404 commented on backend/wordAlgorithm/word.py:
```diff
@@ -0,0 +1,94 @@
+    return ""
+
+# Word printer
+def printWord(word):
```
This should be part of __str__ function in word

Angelo404 commented on backend/wordAlgorithm/word.py:
```diff
@@ -0,0 +1,94 @@
+tree = ET.parse('words.xml');
+root = tree.getroot()
+for child in root:
+    for child1 in child:
```
This loop is confusing

Angelo404 replied:
child and child1 are confusing

Angelo404 replied:
Imports should be at the top. Line 63 to 65 seem copied/pasted if that is the case reference the source in the comments

Angelo404 commented on backend/wordAlgorithm/word.py:
```diff
@@ -0,0 +1,94 @@
+            
+        if child1.tag == "Word_Difficulty":
+            wordDifficulty = child1.text
+    words.append(makeWord(language,originalWord,translatedWord,category,extensions,wordDifficulty))
```
What happens after you store all the words?

Angelo404 commented on backend/wordAlgorithm/wordValueTable.py:
```diff
@@ -0,0 +1,30 @@
+dutchValue = []
+spanishValue = []
+
+import xml.etree.ElementTree as ET
```
Same as the previous file

Angelo404 commented on backend/wordsList/LetterValues/dutch.xml:
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" ?>
```
You might have difficulties with some german letters you might want find different tags for them

Raque10bar committed:
Refactoring

Raque10bar committed:
Fixed mistake

AntonioBarral committed:
Corrected Angelo's corrections

Commit:
added wordValue to resulting wordsList

Raque10bar committed:
Words with translations uploaded according to format

AntonioBarral committed:
Added input (result.xml file needed)

Also Mircea req have been added (i.e. the output will consist on the most challenging sentences and the simplest ones (3 of each)
It is needed to implement how the output will be and solve the problem of having the ip blocked

Raque10bar committed:
Translations working

daangroot committed:
updated letterValues.xml

daangroot committed:
deleted duplicate file

daangroot committed:
added sharp S

FannaIQ commented on backend/wordsList/letterValues.xml:
```diff
@@ -1,575 +1,207 @@
- </Worksheet>
-</Workbook>
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<LetterValues xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
```
I'm sorry but these are not the correct values! the correct ones are in @Raque10bar backend (as explained). These contain more letters for all languages

daangroot committed:
corrected lettervalues

daangroot committed:
corrected first 2500 words

FannaIQ committed:
correct wordObject for xml-format
Does not work in total with the algorithm (main) yet

FannaIQ committed:
correct letter value table with correct XML frequencies
Does not work yet with the algorithm/main

daangroot committed:
corrected last 2500 words

Raque10bar committed:
Removed dutch words duplicates and translated them

Raque10bar committed:
Code updated

FannaIQ committed:
working sorting algorithm

FannaIQ committed:
mistake+tweaking

FannaIQ committed:
sorting with frequency list

Raque10bar committed:
Linguee sentences parser

Raque10bar committed:
Merge pull request #13 from Nickborchers/backendRaquel

Word lists, letter scores and code updated

AntonioBarral committed:
solving conflict error

Raque10bar committed:
Updated Spanish words list

Commit:
Added provisional XML output

Commit:
README added

Commit:
Added structure of the output file

Commit:
Added xml example, refactor algorithm
